drgn/helpers/linux/cgroup.py

```diff
@@ -152,14 +152,14 @@ def css_next_descendant_pre(pos: Object, root: Object) -> Object:
 
 
 def _css_for_each_impl(
-    next_fn: Callable[[Object, Object], Object], css: Object
+    next_fn: Callable[[Object, Object], Object], css: Object, only_online: bool
 ) -> Iterator[Object]:
     pos = NULL(css.prog_, "struct cgroup_subsys_state *")
     while True:
         pos = next_fn(pos, css)
         if not pos:
             break
-        if pos.flags & pos.prog_["CSS_ONLINE"]:
+        if not only_online or pos.flags & pos.prog_["CSS_ONLINE"]:
```

Comment:
If I'm reading it correctly, the kernel code doesn't limit `css_for_each_child()` or `css_for_each_descendant_pre()` to only online css, is that right? If so, then I'd prefer to just remove the `CSS_ONLINE` check entirely, since it's better for helpers to be as similar as possible to the kernel functions they are based on.

Reply:
Moved the onlineness test from helper (to mimic kernel behavior) to selected users.